src/runner.ts

```diff
@@ -0,0 +1,157 @@
+    baseFileName,
+    outputPath
+}: {
+    seedUnits: SourceUnit[];
+    rewrites: Rewrite[];
+    rewriteDepth: number;
+    versions: string[];
+    numberOfTests: number;
+    testCallFunctionName: string;
+    saveVariants: boolean;
+    baseFileName: string;
+    outputPath: string;
+}): Promise<{
+    compilationFailures: Array<[string, CompilationResult]>;
+    fuzzingFailures: Array<[string, string, FuzzingResult]>;
+}> {
+    const compilationFailures: Array<[string, CompilationResult]> = [];
+    const fuzzingFailures: Array<[string, string, FuzzingResult]> = [];
+
+    for (let i = 0; i < numberOfTests; i++) {
+        const compilationResults: CompilationResult[] = [];
+        const fuzzingResults: Array<FuzzingResult | undefined> = [];
+
+        const unit = pickAny(seedUnits);
+        const variant = applyNRandomRewrites(unit, rewrites, rewriteDepth);
+        const variantFileName = path.join(outputPath, `${baseFileName}.variant.${i}.sol`);
+        if (saveVariants) {
+            await saveVariant(variantFileName, variant, versions[0]);
+        }
+        const fuzzer = new Fuzzer(variant, testCallFunctionName, versions[0]);
+        const contractName = unit.vContracts[0].name;
+        for (const version of versions) {
+            const compilationResult = await compile(variant, version);
+            compilationResults.push(compilationResult);
+            if (!compilationResult.success) {
+                // If compilation failed for a specific version, skip the fuzzing
+                continue;
+            }
+
+            const bytecode = compilationResult.bytecode(contractName);
+            if (bytecode === undefined) {
+                fuzzingResults.push(undefined);
+                logger.debug(
+                    `Bytecode not found for contract ${contractName}. Version: ${version}`
+                );
+            } else {
+                const fuzzingResult = await fuzzer.fuzz(bytecode, version);
+                fuzzingResults.push(fuzzingResult);
+            }
+        }
+
+        if (compilationResults.every((result) => !result.success)) {
+            // If all compilations failed, we assume the variant is invalid and skip other checks
+            continue;
+        }
+
+        for (const compilationResult of compilationResults) {
+            if (!compilationResult.success) {
+                // if compilation failed for a specific version, add the variant and version to the list of compilation failures
+                if (!saveVariants) {
+                    // if we're not saving variants, we need to forcefully save the variant file for debugging
+                    await saveVariant(variantFileName, variant, versions[0]);
+                }
+                compilationFailures.push([variantFileName, compilationResult]);
+            }
+        }
+
+        const results = new Set(fuzzingResults.map((result) => result?.runResult.result.result));
+        if (results.size === 1) {
```

Comment:
So if they all succeed then we wont compare the storage? It seems like we still want to check they do the same thing no?

Reply:
Fixed in [93d139d](https://github.com/ConsenSysDiligence/solc-fuzz/pull/3/commits/93d139d70ab85ce749f0f582f9f3e8ef16784f5c)